Tutorial: Update session metrics (#9987)
PostHog/posthog.com: contents/tutorials/session-metrics.md

showTitle: true
author:
  - ian-vanagas
date: 2024-11-26
featuredVideo: 'https://www.youtube-nocookie.com/embed/CVu6ObmOR8Q'
tags:
  - trends
  - sessions
  - product analytics
---

> A session is a set of events grouped to try to capture a "single use" of your product. If you use the [snippet](/docs/getting-started/install?tab=snippet), [JavaScript Web SDK](/docs/libraries/js), or [mobile SDKs](/docs/libraries/ios), PostHog automatically groups events into sessions. We then provide multiple ways to analyze these sessions to get a fuller picture of how users are using your product, where they are spending their time, and more.

Reviewer: ```suggestion
A session is a set of events grouped to capture a "single use" of your product. If you use the [snippet](/docs/getting-started/install?tab=snippet), [JavaScript Web SDK](/docs/libraries/js), or [mobile SDKs](/docs/libraries/ios), PostHog automatically groups events into sessions. We then provide multiple ways to analyze these sessions to get a fuller picture of how users are using your product, where they are spending their time, and more.
```

## Analyzing sessions with web analytics

The easiest way to analyze session metrics is with [web analytics](https://us.posthog.com/web). The [web analytics dashboard](/docs/web-analytics/dashboard) shows you session metrics like:

- Unique session count
- Average session duration
- Bounce rate

You can use filters to get metrics for sessions that match the filters. For example, sessions that visit a specific page or come from a specific referrer.

## Analyzing sessions with product analytics

Beyond the overview web analytics provides, PostHog's [product analytics](/docs/product-analytics/trends/overview) gives you more flexibility to calculate and visualize session metrics. PostHog includes many session properties you might find useful in your analysis like entry and exit URLs, UTMs, pageview count, autocapture count, screen count, web vital LCP score, and more.

### Time on site

Time on site is another name for session duration. Although you can get this web analytics, you can visualize it in a [trend](/docs/product-analytics/trends/overview).

> To do this, first, [create a new insight](https://us.posthog.com/insights/new). Once created, choose pageview as your event, aggregate by property value average of session duration.

Author: Aggregating by property value average is always a bit awkward way from phrasing it. I think since its not _exactly_ what the button in app says, it's fine as is.

> **Tip:** Be sure to set the Y-axis unit in your chart back to `None`.

<ProductScreenshot
    imageLight = "https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_11_26_at_11_35_08_2x_20bea3290f.png"
    imageDark = "https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_11_26_at_11_35_47_2x_8f91b16f3c.png"
    alt="Pages per session" 
    classes="rounded"
/>

You can filter these series to provide details on sections and funnels on your site. Clicking on the charts will provide a list of the users included in that data. If there is a [session replay](/docs/session-replay) available for that user session, there will be a link for you to go watch it.

## Analyzing sessions with SQL

If trends don't give you the details you need, you can access the raw session data with SQL. It is located in the `sessions` table.

For example, to get the average number of unique URLs per session, you can create a new SQL insight and run the following query:

```sql
select avg($num_uniq_urls) from sessions
```

### Connecting sessions to other data

> A big use case for this is combining session data with other event data. Because all events in a `$session_id` property, we can use a `JOIN` to combine session data with event data.

Author: ```suggestion
A big use case for this is combining session data with other event data. Because events have `$session_id` property, we can use a `JOIN` to combine session data with event data.
```
